### PR TITLE
Array/exercise 14

### PR DESCRIPTION
**Pull request by ntnghia0320**

### CODE REVIEW

File: src/main/java/ArrayExercise14.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                    if (j == n - 1) {
+                        j = 0;
+                        i++;
+                    }
+                }
+            }
+        }
+    }
+
+    private static int determinantOfMatrix(int[][] mat, int n) {
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
    private static int determinantOfMatrix(int[][] mat) {
```

File: src/main/java/ArrayExercise14.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        if (n == 1) return mat[0][0];
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
        if (mat.length == 1) return mat[0][0];
```

File: src/main/java/ArrayExercise14.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        int D = 0;
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
        int determinant = 0;
```

File: src/main/java/ArrayExercise14.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+public class ArrayExercise14 {
+    private static void getCofactor(int[][] mat, int[][] temp, int q, int n) {
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
    private static int[][] getCofactor(int[][] source, int q, int n) {
```

File: src/main/java/ArrayExercise14.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    private static int determinantOfMatrix(int[][] mat, int n) {
+        if (n == 1) return mat[0][0];
+
+        int D = 0;
+        int[][] temp = new int[n][n];
+        int sign = 1;
+
+        for (int f = 0; f < n; f++) {
+            getCofactor(mat, temp, f, n);
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
            int[][] temp = getCofactor(mat, f, n);
```

File: src/main/java/ArrayExercise14.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
+public class ArrayExercise14 {
+    private static void getCofactor(int[][] mat, int[][] temp, int q, int n) {
+        int i = 0, j = 0;
+
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
int length = mat.length;
int[][] temp = new int[length  - 1][length  - 1];
```

File: src/main/java/ArrayExercise14.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            for (int col = 0; col < n; col++) {
+                if (row != 0 && col != q) {
+                    temp[i][j++] = mat[row][col];
+                    if (j == n - 1) {
+                        j = 0;
+                        i++;
+                    }
+                }
+            }
+        }
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
        for (int row = 1; row < n; row++) {
            for (int col = 0; col < n; col++) {
                if (col < q) {
                    temp[row - 1][col] = mat[row][col];
                }else if(col > q){
                   temp[row - 1][col - 1] = mat[row][col];
                }
            }
        }
```

File: src/main/java/ArrayExercise14.java (hunk 4 above)
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
    private static void getCofactor(int[][] mat, int[][] temp, int columToRemove, int n) {
```

File: src/main/java/ArrayExercise14.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    private static int determinantOfMatrix(int[][] mat, int n) {
+        if (n == 1) return mat[0][0];
+
+        int D = 0;
+        int[][] temp = new int[n][n];
+        int sign = 1;
+
+        for (int f = 0; f < n; f++) {
```
Comment (voquanghoa) [inline src/main/java/ArrayExercise14.java]:
```suggestion
        for (int column = 0; column < n; column++) {
```

File: src/main/java/ArrayExercise14.java (hunk 1 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 2 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 3 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 4 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 5 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 6 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 7 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 4 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

File: src/main/java/ArrayExercise14.java (hunk 8 above)
Comment (ntnghia0320) [reply src/main/java/ArrayExercise14.java]: Done

**Pull request merged**